test/jdk/tools/sincechecker/SinceChecker.java

```diff
@@ -0,0 +1,769 @@
+                "class: java.lang.reflect.RecordComponent",
+                "class: java.lang.runtime.ObjectMethods",
+                "field: java.lang.annotation.ElementType:RECORD_COMPONENT",
+                "class: java.lang.Record",
+                "method: boolean java.lang.Class.isRecord()",
+                "method: java.lang.reflect.RecordComponent[] java.lang.Class.getRecordComponents()",
+                "field: javax.lang.model.element.Modifier:SEALED",
+                "field: javax.lang.model.element.Modifier:NON_SEALED",
+                "method: javax.lang.model.element.TypeElement:getPermittedSubclasses:()",
+                "method: java.util.List com.sun.source.tree.ClassTree.getPermitsClause()",
+                "method: boolean java.lang.Class.isSealed()",
+                "method: java.lang.constant.ClassDesc[] java.lang.Class.permittedSubclasses()"
+        ));
+
+        LEGACY_PREVIEW_METHODS.put("16", Set.of(
+                "field: jdk.jshell.Snippet.SubKind:RECORD_SUBKIND",
+                "field: javax.lang.model.element.Modifier:SEALED",
+                "field: javax.lang.model.element.Modifier:NON_SEALED",
+                "method: javax.lang.model.element.TypeElement:getPermittedSubclasses:()",
+                "method: java.util.List com.sun.source.tree.ClassTree.getPermitsClause()",
+                "method: boolean java.lang.Class.isSealed()",
+                "method: java.lang.constant.ClassDesc[] java.lang.Class.permittedSubclasses()"
+        ));
+
+        // java.lang.foreign existed since JDK 19 and wasn't annotated - went out of preview in JDK 22
+        LEGACY_PREVIEW_METHODS.put("19", Set.of(
+                "package: java.lang.foreign"
+        ));
+        LEGACY_PREVIEW_METHODS.put("20", Set.of(
+                "package: java.lang.foreign"
+        ));
+        LEGACY_PREVIEW_METHODS.put("21", Set.of(
+                "package: java.lang.foreign"
+        ));
+    }
+
+    private final class EffectiveSourceSinceHelper implements AutoCloseable {
+        private static final JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
+        private final JavaFileManager baseFileManager;
+        private final StandardJavaFileManager fm;
+        private final Set<String> seenLookupElements = new HashSet<>();
+        private final Map<String, Version> signature2Source = new HashMap<>();
+
+        public static EffectiveSourceSinceHelper create(JavacTask mainTask, Collection<? extends Path> sourceLocations, SinceChecker validator) {
+            StandardJavaFileManager fm = compiler.getStandardFileManager(null, null, null);
+            try {
+                fm.setLocationFromPaths(StandardLocation.MODULE_SOURCE_PATH, sourceLocations);
+                return validator.new EffectiveSourceSinceHelper(mainTask, fm);
+            } catch (IOException ex) {
+                try {
+                    fm.close();
+                } catch (IOException closeEx) {
+                }
```

Comment:
Consider using `ex.addSuppressed(closeEx);` instead of just dropping the exception

Reply:
Fixed in 32edb4d811978f87bd2f850c214191e8fc7b8c88